File: backend/app/repositorios/TrabajoSQLiteRepo.go
package repositorios

import(
	"database/sql"
	"mizcuin/modelos"
	"os"
	"fmt"
	_ "github.com/mattn/go-sqlite3"
)

var dbPathTrabajo string

func init(){
	dbPathTrabajo = os.Getenv("DB_PATH")
}

func GuardarTrabajo(consulta string, trabajo modelos.Trabajo) bool{
	db, err := sql.Open("sqlite3", dbPathTrabajo)

	if err != nil {
		fmt.Println("Error al abrir la base de datos en trabajo", err)
		return false
	}

	defer db.Close()

	_, err = db.Exec(consulta, trabajo.Nombre, trabajo.Ubicacion, trabajo.Fecha_publicacion, trabajo.Monto, trabajo.Descripcion, trabajo.Id_usuario, trabajo.Tipo_trabajo, trabajo.Estado)
	if err != nil {
		fmt.Println("Error al ejecutar la consulta en trabajo", err)
		return false
	}

	fmt.Println("Trabajo guardado correctamente: ", consulta)
	return true
	
}

func BuscarIDTrabajo(consulta string, id_trabajo int) (modelos.Trabajo, bool){
	db, err := sql.Open("sqlite3", dbPathTrabajo)

	if err != nil {
		fmt.Println("Error al abrir la base de datos en trabajo", err)
		return modelos.Trabajo{}, false
	}

	defer db.Close()

	row := db.QueryRow(consulta, id_trabajo)

	var idTrabajo int
	var nombre string
	var ubicacion string
	var fechaPublicacion string
	var monto float64
	var descripcion string
	var idUsuario int
	var tipoTrabajo string
	var estado string

	err = row.Scan(&idTrabajo, &nombre, &ubicacion, &fechaPublicacion, &monto, &descripcion, &idUsuario, &tipoTrabajo, &estado)

	if err != nil {
		fmt.Println("Error al buscar el trabajo", err)
		return modelos.Trabajo{}, false
	}

	fmt.Println("Trabajo encontrado correctamente: ", consulta)
	return modelos.Trabajo{Id_trabajo: idTrabajo, Nombre: nombre, Ubicacion: ubicacion, Fecha_publicacion: fechaPublicacion, Monto: monto, Descripcion: descripcion, Id_usuario: idUsuario, Tipo_trabajo: tipoTrabajo, Estado: estado}, true
}

func ActualizarTrabajo(consulta string, trabajo modelos.Trabajo) bool{
	return GuardarTrabajo(consulta, trabajo)
}

func EliminarTrabajo(consulta string, trabajo modelos.Trabajo) bool{
	return GuardarTrabajo(consulta, trabajo)
}

File: backend_Python/app/repositorios/trabajoSQLiterepo.py
import os
import sqlite3

from modelos.trabajo import Trabajo

DB_PATH = os.getenv("DB_PATH")

def guardar_trabajo(consulta, trabajo: Trabajo):
    try:
        conn = sqlite3.connect(DB_PATH)
        cursor = conn.cursor()

        cursor.execute(
            consulta,
            (
                trabajo.nombre,
                trabajo.ubicacion,
                trabajo.fecha_publicacion,
                trabajo.monto,
                trabajo.descripcion,
                trabajo.id_animalLover_publicador,
                trabajo.tipo_trabajo,
                trabajo.estado
            )
        )

        conn.commit()
        conn.close()

        print("Trabajo guardado correctamente:", consulta)
        return True

    except Exception as e:
        print("Error al ejecutar la consulta en trabajo:", e)
        return False
    
def buscar_trabajo_por_id(consulta, id_trabajo):
    try:
        conn = sqlite3.connect(DB_PATH)
        cursor = conn.cursor()

        cursor.execute(consulta, (id_trabajo,))
        fila = cursor.fetchone()
        conn.close()

        if fila is None:
            return None, False

        trabajo = Trabajo(
            fila[0],  # id_trabajo
            fila[1],  # nombre
            fila[2],  # ubicacion
            fila[3],  # fecha_publicacion
            fila[4],  # monto
            fila[5],  # descripcion
            fila[6],  # id_animalLover_publicador
            fila[7],  # tipo_trabajo
            fila[8]   # estado
        )

        print("Trabajo encontrado correctamente:", consulta)
        return trabajo, True

    except Exception as e:
        print("Error al buscar el trabajo:", e)
        return None, False
    
def actualizar_trabajo(consulta, trabajo):
    return guardar_trabajo(consulta, trabajo)

def eliminar_trabajo(consulta, trabajo):
    return guardar_trabajo(consulta, trabajo)